super_dpf.py

```diff
@@ -1,127 +1,316 @@
-        if not os.path.exists(self.PATHS['photos']):
-            os.makedirs(self.PATHS['photos'])
+    def gphotos_configure_dialogue(self):
+        # https://picasaweb.google.com/data/feed/base/user/107119165375640000020/albumid/6225063173331304033?alt=rss&kind=photo&authkey=Gv1sRgCPzC_v7n98aRmgE&hl=en_US
+        feed_url = self._parse_feed_url(raw_input('Google Photos Feed URL: '))
+        user_id = raw_input('Google Photos Username: ')
+        album_id = raw_input('Album ID: ')
+        enable = raw_input('Enter "yes" to enable Google Photo sync: ')
+
+        enable = 'true' if enable.lower() == 'yes' else ''
+
+        if not os.path.isdir(self.gphotos_dir):
+            os.makedirs(self.gphotos_dir)
+
+        self._save_settings(g_enabled=enable, feed_url=list(feed_url),
+                            user_id=user_id, album_id=album_id)
+
+    def _gphotos_sync(self):
+        if not self.gphotos_configured:
+            logging.info('Google Photos is not configured.')
+            configure = raw_input('Configure it now? ')
+            if configure.lower() == 'yes':
+                self.gphotos_configure_dialogue()
+        elif not self.gphotos_enabled:
+            logging.info('Google Photos is not enabled.')
+            enable = raw_input('Enable it now? ')
+            enable = 'true' if enable.lower() == 'yes' else ''
+            self._save_settings(g_enabled=enable)
+        else:
+            feed_url = self.settings['gphotos'].get('feed_url')
+            for feed in feed_url:
+                response = requests.get(feed)
+                tags = BeautifulStoneSoup(
+                    response.content).findAll('media:content')
+                for tag in tags:
+                    if tag.get('url', False):
+                        filename = os.path.join(
+                            self.gphotos_dir,
+                            os.path.basename(urlsplit(tag['url'])[2]))
+                        if not os.path.isfile(filename):
+                            logging.info(
+                                'Getting img at {}'.format(tag['url']))
+                            response = requests.get(tag['url'], stream=True)
+                            try:
+                                img = Image.open(StringIO(response.content))
+                                img.save('{}'.format(filename), optimize=True,
+                                         progressive=True, quality=100,
+                                         subsampling=0)
+                            except IOError as e:
+                                logging.error(e)
+
+
+class DropboxController(BaseDPF):
+    pass
+
+
+class FlickrController(BaseDPF):
+    # https://stuvel.eu/media/flickrapi-docs/documentation/
+    pass
+
+
+class InstagramController(BaseDPF):
+    # https://github.com/Instagram/python-instagram
+    pass
+
+
+class HotMediaController(BaseDPF):
+    pass
+
+
+class SuperDPF(AWSController, GPhotoController):
```

Comment:
In the context of my other comments this class would not exist as a subclass of the other to classes.  As more subclasses are created this will become dangerous to change due to the increasing odds of namespace collisions across all of the DPF subclasses.

I would instead suggest that this class be responsible for handling run arguments that put it in a configuration mode or a sync mode.  In configuration mode it could maintain a list of "Accounts" (I'm thinking of the multiple account idom on android) where each account pertains to a specific subclass of BaseDPF.  This would allow for multiple accounts of the same type and it would obviate the need for marking each type as "enabled".  Each account object would contain the class name and data returned by the config dialog classmethod I mention in another comment.  When it is time to sync you can then iterate over this list of account objects instantiating the class with the account details and calling the sync() method.